src/main/kotlin/com/openlattice/mechanic/upgrades/MigrateOrgPermissionsUpgrade.kt

```diff
@@ -0,0 +1,127 @@
+package com.openlattice.mechanic.upgrades
+
+import com.hazelcast.query.Predicates
+import com.openlattice.authorization.Ace
+import com.openlattice.authorization.AceKey
+import com.openlattice.authorization.AceValue
+import com.openlattice.authorization.Acl
+import com.openlattice.authorization.Action
+import com.openlattice.authorization.mapstores.PermissionMapstore
+import com.openlattice.authorization.securable.SecurableObjectType
+import com.openlattice.hazelcast.HazelcastMap
+import com.openlattice.mechanic.Toolbox
+import com.openlattice.postgres.external.ExternalDatabaseConnectionManager
+import com.openlattice.postgres.external.ExternalDatabasePermissioningService
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+import java.sql.SQLException
+import java.util.UUID
+import java.util.regex.Pattern
+
+class MigrateOrgPermissionsUpgrade(
+        toolbox: Toolbox,
+        private val exConnMan: ExternalDatabaseConnectionManager,
+        private val exDbPermMan: ExternalDatabasePermissioningService
+): Upgrade {
+
+    val logger: Logger = LoggerFactory.getLogger(MigrateOrgPermissionsUpgrade::class.java)
+
+    private val entitySets = HazelcastMap.ENTITY_SETS.getMap(toolbox.hazelcast)
+    private val externalTables = HazelcastMap.EXTERNAL_TABLES.getMap(toolbox.hazelcast)
+    private val permissions = HazelcastMap.PERMISSIONS.getMap(toolbox.hazelcast)
+
+    override fun upgrade(): Boolean {
+        val uuidRegex= Pattern.compile(
+            "[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}",
+            Pattern.CASE_INSENSITIVE
+        )
```

Comment:
This seems unnecessary. It's easier to just do:

```
try {
  orgId = UUID.fromString(input)
} catch (e: Exception) {
  logger.error(e)
  return
}
```

Reply:
I was initially thinking that this looks (cosmetically) better, but it is unnecessary + it doesn't just error out if the input is not a UUID. I will change it back.